src/ApiGatewayThrottlingSettings.js

```diff
@@ -28,11 +28,16 @@ class ApiGatewayEndpointThrottlingSettings {
       this.method = event.http.method;
     }
 
-    this.maxRequestsPerSecond = get(event.http.throttling, 'maxRequestsPerSecond', globalSettings.maxRequestsPerSecond);
-    this.maxConcurrentRequests = get(event.http.throttling, 'maxConcurrentRequests', globalSettings.maxConcurrentRequests);
+    if (event.http.throttling.isEmpty) {
+      this.maxRequestsPerSecond = -1
+      this.maxConcurrentRequests = -1
+    }
+    else {
+      this.maxRequestsPerSecond = get(event.http.throttling, 'maxRequestsPerSecond', globalSettings.maxRequestsPerSecond);
+      this.maxConcurrentRequests = get(event.http.throttling, 'maxConcurrentRequests', globalSettings.maxConcurrentRequests);
+    }
```

Comment:
Hi @scadu,

This code is using [lodash.get](https://lodash.com/docs/4.17.15#get) which has the signature:
```
get(object, path, [defaultValue])
```

So when `event.http.throttling` is not defined, it uses the default value. That means the code can be simplified:

```
this.maxRequestsPerSecond = get(event.http.throttling, 'maxRequestsPerSecond', -1);
this.maxConcurrentRequests = get(event.http.throttling, 'maxConcurrentRequests', -1);
```

Also I think `globalSettings` don't need to be passed into this constructor anymore, since this was the only place that was using it.

Reply:
Good point, I'll remove those conditionals. I just started tinkering with JS and at the time of doing this I wasn't sure how to define a default for constructor, since I was mislead thinking it was a multi-parameter constructor.

I'm looking into tests at the moment.